packages/trader/src/Modules/Trading/Components/Form/ContractType/ContractTypeMenu/contract-type-menu.jsx

```diff
@@ -8,111 +8,97 @@ import NoResultsMessage from './no-results-message.jsx';
 import { Header } from '../ContractTypeInfo';
 import { getContractCategoryLabel } from '../../../../Helpers/contract-type';
 
-class Dialog extends React.PureComponent {
-    input_ref = React.createRef();
+const Dialog = ({
+    categories,
+    item,
+    selected,
+    children,
+    is_info_dialog_open,
+    is_open,
+    onBackButtonClick,
+    onCategoryClick,
+    onChangeInput,
+}) => {
+    const input_ref = React.useRef(null);
 
-    state = {
-        input_value: '',
-    };
+    const [input_value, setInputValue] = React.useState('');
+
+    const contract_category = getContractCategoryLabel(categories, item);
+    const selectedItem = selected ? { label: selected } : contract_category;
```

Comment:
Can we use snake_case here too? `selected_item`

Reply:
yes, sure